gnucash/gnome-search/search-date.c

```diff
@@ -178,10 +178,23 @@ gncs_validate (GNCSearchCoreType *fe)
     return valid;
 }
 
+static time64
+adjust_date_value (GNCSearchDate *fi, time64 time_in)
+{
+    /* time64 time_in comes from the gnc_date_edit_get_date which
+     * returns a value set to the start of the day 00:00:00, adjust
+     * value to 23:59:59 for LessThanEqual and GreaterThan */
+
+    if (fi->how == QOF_COMPARE_LTE || fi->how == QOF_COMPARE_GT)
+        return gnc_time64_get_day_end (time_in);
+    else
+        return time_in;
+}
+
```

Comment:
Good.

Should we anticipate that gnc_date_edit_get_date may one day be changed to return 10:59:59 as time (which is our new default neutral time) ? That is to say this function may be more robust if it also acted on QOF_COMPARE_LT and QOF_COMPARE_GTE to return gnc_time64_get_day_begin.

Reply:
Better:
```C
static void
gnc_search_date_set_date_from_edit(GncSearchDate* fe, GncDateEdit* ed)
{
    if (fi->how == QOF_COMPARE_LTE || fi->how == QOF_COMPARE_GT)
        fe->tt = gnc_date_edit_get_date_end (ed);
    else
       fe->tt = gnc_date_edit_get_date (ed);
}

static void
date_changed (GncDateEdit *date_edit, GNCSearchDate *fe)
{
    gnc_search_date_set_date_from_edit (fe, date_edit);
}
```
and same for gncs_get_predicate. ```gnc_date_edit_get_date_end``` sets the time to midnight using the ```struct tm``` from ```gnc_date_edit_get_date_internal``` so it will be faster than gnc_time64_get_day_end's round-trip into a GncDateTime and it clearly expresses the intent for future refactoring of gnc_date_edit.